src/InputAreaViewModel.ts

```diff
@@ -0,0 +1,405 @@
+    name: 'collapsed',
+    notify: InputAreaViewModel.stateChangedSignal,
+  });
+
+  /**
+  * A property descriptor containing the prompt number.
+  *
+  * **See also:** [[promptNumber]]
+  */
+  static promptNumberProperty = new Property<InputAreaViewModel, number>({
+    name: 'promptNumber',
+    notify: InputAreaViewModel.stateChangedSignal,
+  });
+
+  /**
+  * A property descriptor containing the execution count of the input area.
+  *
+  * **See also:** [[executionCount]]
+  */
+  static executionCountProperty = new Property<InputAreaViewModel, number>({
+    name: 'executionCount',
+    notify: InputAreaViewModel.stateChangedSignal,
+  });
+
+  /**
+   * A signal emitted when the state of the model changes.
+   *
+   * #### Notes
+   * This is a pure delegate to the [[stateChangedSignal]].
+   */
+  get stateChanged() {
+    return InputAreaViewModel.stateChangedSignal.bind(this);
+  }
+
+  /**
+   * Get whether the input area should be a rendered representation.
+   *
+   * #### Notes
+   * This is a pure delegate to the [[renderedProperty]].
+   */
+  get rendered() {
+    return InputAreaViewModel.renderedProperty.get(this);
+  }
+
+  /**
+   * Set whether the input area should be a rendered representation.
+   *
+   * #### Notes
+   * This is a pure delegate to the [[renderedProperty]].
+   */
+  set rendered(value: boolean) {
+    InputAreaViewModel.renderedProperty.set(this, value);
+  }
+
+
+  /**
+   * Get whether the input area should be collapsed or displayed.
+   *
+   * #### Notes
+   * This is a pure delegate to the [[collapsedProperty]].
+   */
+  get collapsed() {
+    return InputAreaViewModel.collapsedProperty.get(this);
+  }
+
+  /**
+   * Set whether the input area should be collapsed or displayed.
+   *
+   * #### Notes
+   * This is a pure delegate to the [[collapsedProperty]].
+   */
+  set collapsed(value: boolean) {
+    InputAreaViewModel.collapsedProperty.set(this, value);
+  }
+
+  /**
+   * Get the prompt number.
+   *
+   * #### Notes
+   * This is a pure delegate to the [[promptNumberProperty]].
+   */
+  get promptNumber() {
+    return InputAreaViewModel.promptNumberProperty.get(this);
+  }
+
+  /**
+   * Set the prompt number.
+   *
+   * #### Notes
+   * This is a pure delegate to the [[promptNumberProperty]].
+   */
+  set promptNumber(value: number) {
+    InputAreaViewModel.promptNumberProperty.set(this, value);
+  }
+
+  /**
+   * Get the execution count of the input area.
+   *
+   * #### Notes
+   * This is a pure delegate to the [[executionCountProperty]].
+   */
+  get executionCount() {
+    return InputAreaViewModel.executionCountProperty.get(this);
+  }
+
+  /**
+   * Set the execution count of the input area.
+   *
+   * #### Notes
+   * This is a pure delegate to the [[executionCountProperty]].
+   */
+  set executionCount(value: number) {
+    InputAreaViewModel.executionCountProperty.set(this, value);
+  }
+
+    textEditor: ITextEditorViewModel;
```

Comment:
In general, I'd say that to keep things simple, if you don't need a getter/setter, don't use one.  However, in this case, we need to trigger an update signal so that the widgets know that that whole part of the state changed.  My bad not realizing this at first.  Thanks.